view/frontend/templates/authorization.phtml

```diff
@@ -62,32 +59,42 @@ $checkoutKey = $block->getCheckoutKey();
             scriptTag.setAttribute('data-publishable-key', '<?= /* @noEscape */ $checkoutKey; ?>');
             document.head.appendChild(scriptTag);
         }
+
+        let injectButtonsWhenScriptLoaded = function() {
```

Comment:
```suggestion
        window.whenDefined(window, 'BoltAccount', function () {
            BoltAccount.injectButtons(); // optional since BoltAccount calls injectButtons onDOMContentLoaded and observer will also trigger for the first one
            /**
             * Triggers {@see BoltAccount.injectButtons} whenever a new Bolt login button is injected into the DOM
             * currently happens in the default Magento theme to facilitate mobile menu
             */
            domObserver.get('div.bolt-account-sso', function () {
                    BoltAccount.injectButtons();
            }); //this can optionally be limited to only observe header
        });
```
this replaces the rest of the JS logic and will trigger on any login button insertions (put here as a reference you'll need to do some refactoring)